src/bun.js/bindings/bindings.zig

```diff
@@ -1802,6 +1807,106 @@ pub const JSInternalPromise = extern struct {
     };
 };
 
+const AnyPromiseTag = enum { Normal, Internal };
+pub const AnyPromise = union(AnyPromiseTag) {
+    Normal: *JSPromise,
+    Internal: *JSInternalPromise,
+
+    pub fn status(this: AnyPromise, vm: *VM) JSPromise.Status {
+        switch (this) {
+            AnyPromiseTag.Normal => {
+                return this.Normal.status(vm);
+            },
+            AnyPromiseTag.Internal => {
+                return this.Internal.status(vm);
+            },
+        }
+    }
+    pub fn result(this: AnyPromise, vm: *VM) JSValue {
+        switch (this) {
```

Comment:
1) `switch` can be an expression
2) Tag can be ommitted when it knows what type to expect, just an enum literal
3) `union(enum)` can be used to avoid the explicit tag
```zig
return switch(this) {
  .Normal => |promise| promise.result(vm),
 .Internal => |promise| promise.result(vm)
```

Furthermore, the new feature in Zig we can use here is `inline` and we get an even shorter thing:
```zig
return switch (this) {
   inline else => |promise| promise.result(vm)
};
```

Reply:
Thanks for the tips - done in https://github.com/oven-sh/bun/pull/1697/commits/0323676611ecdefe46d6e334d52dfc7360b2856e